firebase-database/src/androidTest/java/com/google/firebase/database/QueryTest.java

```diff
@@ -4554,16 +4581,22 @@ public void getUpdatesPersistenceCacheWhenEnabled()
     FirebaseDatabase readerDb = FirebaseDatabase.getInstance(readerApp);
     readerDb.setPersistenceEnabled(true);
     FirebaseDatabase writerDb = FirebaseDatabase.getInstance(writerApp);
+
     DatabaseReference reader = readerDb.getReference();
     DatabaseReference writer = writerDb.getReference();
+    String key = reader.push().getKey();
+    reader = reader.child(key);
+    writer = writer.child(key);
 
     assertNull(new WriteFuture(writer, 42L).timedGet());
     assertEquals(42L, Tasks.await(reader.get()).getValue());
 
     readerDb.goOffline();
-
-    Semaphore semaphore = new Semaphore(0);
-    assertNotNull(ReadFuture.untilEquals(reader, 42L).timedGet());
+    try {
+      assertNotNull(ReadFuture.untilEquals(reader, 42L).timedGet());
+    } finally {
+      readerDb.goOnline();
+    }
```

Comment:
Can you add a test that verifies that we don't go back to the `SingleValueEvent` behavior. That is:

- Client 1: Read value:old from cache
- Client 1: Disconnect all listeners
- Client 2: Write value:new
- Client 1:  Get should now return value:new